app/models/auth_sources/auth_source_ldap.rb

```diff
@@ -130,6 +131,28 @@ def users_in_group(name)
     ldap_con.user_list(name)
   end
 
+  def test_connection
+    result = {}
+    begin
+      Timeout::timeout(10) do
+        self.ldap_con.valid_user?("")
+      end
+      result[:success] = true
+      result[:message] = _("Test connection to LDAP server was successful.")
+    rescue Timeout::Error, Net::LDAP::Error  => exception
+      exception = Foreman::Exception.new exception.message
+      result[:success] = false
+      result[:error_class] = exception.class.name
+      result[:message] = _(exception.message)
```

Comment:
Why? That's useful when people want to debug what's going on, they can just take the error message and search on the internet what happened - we even keep track of possible solutions to each error code [on the wiki](http://projects.theforeman.org/projects/foreman/wiki/ErrorCodes)